ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckAnalyzer.java

```diff
@@ -63,13 +67,24 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     }
 
     Table table = getTable(tableName);
-    List<Map<String, String>> specs = getPartitionSpecs(table, root);
+    Map<Integer, List<ExprNodeGenericFuncDesc>> partitionSpecs = getFullPartitionSpecs(root, table, conf, false);
+    byte[] filterExp = null;
+    if (partitionSpecs != null & !partitionSpecs.isEmpty()) {
+      // explicitly set expression proxy class to PartitionExpressionForMetastore since we intend to use the
+      // filterPartitionsByExpr of PartitionExpressionForMetastore for partition pruning down the line.
+      conf.set(MetastoreConf.ConfVars.EXPRESSION_PROXY_CLASS.getVarname(),
+          PartitionExpressionForMetastore.class.getCanonicalName());
+      // fetch the first value of partitionSpecs map since it will always have one key, value pair
+      filterExp = SerializationUtilities.serializeExpressionToKryo(
```

Comment:
why this needs to be flattened into a `byte[]` ?

Reply:
PartitionPruner and msc.listPartitionsByExpr() expects serialized byte array, Hence it is required to flatten out.